crypto/bls.go

```diff
@@ -179,6 +179,17 @@ func (a *blsBLS12381Algo) generatePrivateKey(seed []byte) (PrivateKey, error) {
 	return sk, nil
 }
 
+// BLSInvalidSignature returns an invalid signature that fails when verified
+// with any message and public key.
+//
+// The signature bytes represent an invalid serialization of a point which
+// makes the verification fail early.
+func BLSInvalidSignature() Signature {
+	signature := make([]byte, SignatureLenBLSBLS12381)
+	signature[0] = 0xC1 // invalid header as per C.ep_read_bin_compact
```

Comment:
Nit:
This works, but you have to dig into step 4 of the [deserialization spec](https://www.ietf.org/archive/id/draft-irtf-cfrg-pairing-friendly-curves-10.html#name-point-deserialization-proce) to see why, and we're 1 bit away from returning infinity.

If we made a mistake in our code by forgetting [this if branch](https://github.com/onflow/flow-go/blob/dd75d07c09a5e957ef6a7adae898f95c67775054/crypto/bls12381_utils.c#L304-L307) we'd be in trouble.

If we put in `0x20`, `0x60`, or `0xE0`, we fall into the explicitly invalid cases of the Zcash spec just on the metadata bits, without having to inspect the lower bits of the byte :
```
If m_byte equals any of 0x20, 0x60, or 0xE0, output INVALID and stop decoding.
```

Moreover, if we put in `0x20`, the de-serializer would have to be pretty darn "creative" to deserialize to the infinity, and therefore has very little chance to add this to anything and preserve any sort of validity.

Reply:
> if we made a mistake in our code by forgetting this if branch we'd be in trouble.

That's what tests are for :) 

In our implementation, any byte with an MSB equal to `0` makes the implementation fail very early (both in code and spec steps), because we only support compressed points. I wanted to use a byte that starts with `1`, so that we don't have to update this byte in the future if we were to support uncompressed points (that's definitely not a strong argument). 
We're left with `0xE0`. It actually fails exactly at the same [step](https://github.com/onflow/flow-go/blob/dd75d07c09a5e957ef6a7adae898f95c67775054/crypto/bls12381_utils.c#L306) as `0xC1` (looking at the code steps not the spec steps). I don't mind updating to `0xE0` though.

I believe these are very small details, the relevant point to optimize for is to not enter the membership check or the pairing when verifying.